plumbing/object/file.go

```diff
@@ -81,7 +81,7 @@ func (iter *FileIter) Next() (*File, error) {
 			return nil, err
 		}
 
-		if entry.Mode.IsDir() {
+		if entry.Mode.IsDir() || entry.Mode == SubmoduleMode {
```

Comment:
Now, this is how it works:

The `FileIter.Next` method returns `Files` by going over all the elements in the tree, ignoring Dirs, and now (added in this PR) Submodules.

This is what I propose, instead:

The `FileIter.Next` method returns `Files` by going over all the elements in the tree, ignoring everything that is not a file (without naming explicitly what are not files).

This way we don't have to change the code everytime we support a new TreeEntry mode.

So instead of:
```
if entry.Mode.IsDir() || entry.Mode == SubmoduleMode {
    continue
}
```

I suggest:

```
if !entry.Mode.IsFile() && !entry.Mode.IsOldFileGroupWriteable() {
    continue
}
```